modules/@apostrophecms/schema/index.js

```diff
@@ -2190,9 +2190,13 @@ module.exports = {
         }
         _.each(patch, function(val, key) {
           if (key.charAt(0) !== '$') {
-            key = self.apos.util.resolveAtReference(existingPage, key);
+            let atReference = false;
+            if (key.charAt(0) === '@') {
```

Comment:
This was the bug that broke undo. Apostrophe has a sweet extension to PATCH to let you patch any sub-object by the _id of that nested object using this syntax: `@id-goes-here-` but it was broken for the edge case where the sub-object you're replacing is a top level property of the page, like the "main" area. Simple to fix.